### ISSUE TITLE
CPU Parallelization

### ISSUE THREAD

**Issue by syndamn** [DiffEqML/torchdyn#115]
Hello, I was wondering if there was any way to do paralellization with cpu cores to train using this package? When I've tried to implement torch multiprocessing, it warns me about a fork and autograd conflict.

**Comment by Zymrael** [comment]
This all depends on what you'd like to do specifically. PyTorch already parallelizes ops across CPU cores afaik. Are you trying to train **different** models across **different** CPU nodes?

Could you also share a minimal example?

**Comment by Zymrael** [comment]
Not sure what the use case would be, as all methods and classes here can already solve in parallel batches of initial conditions. Are you looking for more fine-grained control on which initial conditions go on which core?

**Comment by syndamn** [comment]
Yes! It would also be super helpful if you could explain how the parallel batching works. When  using the process above, is the model training on three (or some  number) separate initial conditions simultatneously and developing the weights for the model based on the combined loss of the multiple initial conditions? I just want to verify if it is possible to train the model using multiple ICs with each IC set having access to the learning results of the others. 

Hopefully that makes sense, thank you for the help :)

**Comment by Zymrael** [comment]
Sure, the different ICs here represent your batch of data in a machine learning sense. So your tensor dimension dedicated to batching will carry forward your solutions for each IC. This means that if needed you can use these values anytime during the training procedure (to compute for example normalization statistics) or only at the end to compute the mean value of some target loss function.

CPU parallelization of data samples within a batch is not a `torchdyn` specific feature and is due to `PyTorch`. You can find plenty of in-depth information online if you need to dig into its implementation further.
